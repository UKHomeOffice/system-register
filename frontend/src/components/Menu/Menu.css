.Menu{
    background-color: #F8F8F8;
    min-width: 100%;
}

.Menu li{
    display: inline-block;
    margin-right: 20px;
    font-size: larger;
    font-weight: bold;
    height: 40px;
}

.Menu ul{
    list-style: none;
    padding-inline-start: 0;
    margin-top: 0;
    padding-top: 12px;
}

.Menu a:visited{
    color: #1E70B8;
}

.Menu a:focus-visible {
    text-decoration: none;
    text-decoration-color: transparent;
    outline: none;
    background-color: var(--yellow);
    border-bottom: 5px solid var(--black);
    padding-bottom: 0;
}

.Menu a:hover{
    text-decoration: underline;
}

.Menu a:active {
    text-decoration: none;
    outline: none;
    background-color: var(--yellow);
    border-bottom: 5px solid var(--black);
    padding-bottom: 0;
}

.Menu a.selected:after{
    content: "";
    border-bottom: 5px solid #1d70b8;
    width: 100%;
    float: right;
    padding-top: 12px;
}

.Menu a{
    color: #1E70B8;
    padding-bottom: 12px;
    text-decoration: none;
}