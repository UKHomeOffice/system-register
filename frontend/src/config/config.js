import axios from "axios";

const baseUrl = process.env.NODE_ENV === "development"
  ? "http://localhost:8080"
  : "";

const config = {
  baseUrl,
  getKeycloakConfig,
  api: {
    url: `${baseUrl}/api`
  },
};

async function getKeycloakConfig() {
  const res = await axios.get(`${baseUrl}/config/keycloak`);
  return {
    url: res.data.host,
    realm: res.data.realm,
    clientId: res.data.clientId,
  };
}

export default config;
