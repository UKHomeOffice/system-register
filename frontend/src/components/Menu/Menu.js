import React from 'react'
import { NavLink } from 'react-router-dom'
import './Menu.css'

const Menu = () => {
    return (
        <nav className="Menu">
            <ul className="centerContent">
                <li><NavLink exact to="/" activeClassName="selected">System Register</NavLink></li>
                <li><NavLink to="/risk_dashboard" activeClassName="selected">Risk Dashboard</NavLink></li>
                <li><NavLink to="/about" activeClassName="selected">About</NavLink></li>
                <li><NavLink to="/contact" activeClassName="selected">Contact</NavLink></li>
            </ul>
        </nav>
    )
}

export default Menu