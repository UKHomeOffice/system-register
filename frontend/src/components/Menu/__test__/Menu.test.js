import React from 'react'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'
import { BrowserRouter, Router, MemoryRouter } from 'react-router-dom'
import Menu from '../Menu'
import config from "../../../config/config";


describe('<Menu />', () => {
    it('renders System Register link', () => {
        const { getByText } = render(<MemoryRouter><Menu /></MemoryRouter>)
        const element = getByText('System Register')
        expect(element).toBeInTheDocument()
        expect(element).toHaveAttribute("href", '/')
    });

    it('renders Risk Dashboard link', () => {
        const { getByText } = render(<MemoryRouter><Menu /></MemoryRouter>)
        const element = getByText('Risk Dashboard')
        expect(element).toBeInTheDocument()
        expect(element).toHaveAttribute("href", '/risk_dashboard')
    });

    it('renders API link', () => {
        const { getByText } = render(<MemoryRouter><Menu /></MemoryRouter>)
        const element = getByText('API')
        expect(element).toBeInTheDocument()
        expect(element).toHaveAttribute("href", `${config.api.url}/systems`)
    });

    it('renders About link', () => {
        const { getByText } = render(<MemoryRouter><Menu /></MemoryRouter>)
        const element = getByText('About')
        expect(element).toBeInTheDocument()
        expect(element).toHaveAttribute("href", '/about')
    });

    it('renders Contact link', () => {
        const { getByText } = render(<MemoryRouter><Menu /></MemoryRouter>)
        const element = getByText('Contact')
        expect(element).toBeInTheDocument()
        expect(element).toHaveAttribute("href", '/contact')
    });

    describe('when selecting Systems Register page', () => {
        it('should identify you are situated in the System Register page', () => {
            const history = createMemoryHistory()
            const route = '/'
            history.push(route)
            const { getByText } = render(<Router history={history}><Menu /></Router>)
            const element = getByText('System Register')
            expect(element).toHaveClass('selected')
        });

        it('should not identify you are situated in the Risk Dashboard page', () => {
            const history = createMemoryHistory()
            const dashboard_route = '/risk_dashboard'
            const home_route = '/'
            history.push(dashboard_route)
            history.push(home_route)
            const { getByText } = render(<Router history={history}><Menu /></Router>)
            const registerElement = getByText('System Register')
            const dashboardElement = getByText('Risk Dashboard')
            expect(registerElement).toHaveClass('selected')
            expect(dashboardElement).not.toHaveClass('selected')
        });
    })

    describe('when selecting Systems Register page', () => {
        it('should identify you are situated on the risk dashboard page', () => {
            const history = createMemoryHistory()
            const route = '/risk_dashboard'
            history.push(route)
            const { getByText } = render(<Router history={history}><Menu /></Router>)
            const element = getByText('Risk Dashboard')
            expect(element).toHaveClass('selected')
        })
    })

    it('matches snapshot', () => {
        const snapshot = renderer.create(<BrowserRouter><Menu /></BrowserRouter>)
        expect(snapshot).toMatchSnapshot()
    });
})