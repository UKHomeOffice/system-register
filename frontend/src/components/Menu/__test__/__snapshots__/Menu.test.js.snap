// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches snapshot 1`] = `
<nav
  className="Menu"
>
  <ul
    className="centerContent"
  >
    <li>
      <a
        aria-current="page"
        className="selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        System Register
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/risk_dashboard"
        onClick={[Function]}
      >
        Risk Dashboard
      </a>
    </li>
    <li>
      <a
        href="/api/systems"
      >
        API
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
